src/lib.rs

```diff
@@ -587,39 +587,55 @@ where
     pub async fn take(
         mut self,
     ) -> Result<impl Stream<Item = Result<Entry<'scan, R>, ParquetError>>, DbError<R>> {
-        self.streams.push(
-            self.schema
-                .mutable
-                .scan((self.lower, self.upper), self.ts)
-                .into(),
-        );
-        for (_, immutable) in self.schema.immutables.iter() {
-            let (lower, upper) = if self.reversed == Some(true) {
+        // Helper function to handle reversing logic
+        fn maybe_reverse<T>(vec: &mut Vec<T>, reverse: bool) {
+            if reverse {
+                vec.reverse();
```

Comment:
reverse the vector of merge stream dooes not make sense, the order of merge stream should keep `mutable memtable -> immutable memtable -> leveled sstable` to ensure read the value with highest timestamp.

Reply:
understood.